metagraph/src/graph/graph_extensions/node_rc.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef __NODE_RC_HPP__
+#define __NODE_RC_HPP__
+
+#include <sdsl/dac_vector.hpp>
+#include <sdsl/rrr_vector.hpp>
+
+#include "graph/representation/base/sequence_graph.hpp"
+#include "common/vectors/bit_vector_adaptive.hpp"
+
+namespace mtg {
+namespace graph {
+
+// Maps each node in a PRIMARY-mode DeBruijnGraph to nodes adjacent to its reverse
+// complement. When using the index construct constructor on DBGSuccinct, or when
+// loading such an index, this stores a map from each DBGSuccinct to the BOSS nodes
+// corresponding to the reverse complements of its k-1 prefix and k-1 suffix.
+class NodeRC : public SequenceGraph::GraphExtension {
+  public:
+    using node_index = SequenceGraph::node_index;
+
+    // If construct_index is false, then an index must be loaded to take advantage
+    // of the index for call_*_from_rc calls
```

Comment:
```suggestion
    // If construct_index is true, a fully pre-computed index will be constructed.
    // Otherwise, call_*_from_rc will invoke on-the-fly computations on the graph,
    // unless the index is loaded with load().
```